### PR TITLE
Add contributing doc

### PR DESCRIPTION
**Pull request by ericglau**
Add basic contributing guidelines including how to build/test the project locally.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+Contributions to OpenZeppelin Upgrades are welcome. Please review the information below to test out and contribute your changes.
+
+## Building and testing the project
+
+### Prerequisites
+The following prerequisites are required to build the project locally:
+- [Node.js](https://nodejs.org/)
+- [Yarn](https://yarnpkg.com/getting-started/install)
+
+After the prerequisites are installed, the commands below can be run from this project's `openzeppelin-upgrades` directory.
```
Comment (frangio) [inline CONTRIBUTING.md:13]:
```suggestion
After the prerequisites are installed, the commands below can be run from this project's root directory.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+### Building the project
```
Comment (frangio) [inline CONTRIBUTING.md:15]:
```suggestion
### Installing dependencies
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+```yarn install```
+
+The project must be built at least once before running the tests or linter.
```
Comment (frangio) [inline CONTRIBUTING.md:18]: I don't think this is true. Can you check?

Comment (ericglau) [reply CONTRIBUTING.md]: On a clean project without running `yarn install` first, the following occur:

yarn test:
```
yarn run v1.22.15
$ tsc -b && wsrun -ms test
/bin/sh: tsc: command not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

yarn lint:
```
yarn run v1.22.15
$ yarn lint:path .
$ eslint --ignore-path .gitignore --max-warnings 0 .
/bin/sh: eslint: command not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Review by Amxx — APPROVED** [review]

**Pull request merged**